File: app/src/types/voice-file.ts
export interface VoiceFile {
  id: number
  user_id: number
  voice_role?: string
  voice_url: string
  duration?: number
  file_size?: number
  created_at: string
  updated_at: string
}

export interface CreateVoiceFileRequest {
  user_id: number
  voice_role?: string
  voice_url: string
  duration?: number
  file_size?: number
}

export interface UpdateVoiceFileRequest {
  user_id?: number
  voice_role?: string
  voice_url?: string
  duration?: number
  file_size?: number
}

export interface GetVoiceFilesParams {
  user_id?: number
  page?: number
  size?: number
}

export interface GetVoiceFilesResponse {
  data: VoiceFile[]
}

export interface DeleteVoiceFileResponse {
  message: string
}

export interface GetVoiceFileContentResponse {
  url?: string
}

File: api/models/voice_file.go
package models

import "time"

type VoiceFile struct {
	ID        int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID    int64     `gorm:"index;not null" json:"user_id"`
	VoiceRole string    `gorm:"size:64" json:"voice_role"`
	VoiceURL  string    `gorm:"size:512;not null" json:"voice_url"`
	Duration  int       `gorm:"default:0" json:"duration"`
	FileSize  int64     `gorm:"default:0" json:"file_size"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (VoiceFile) TableName() string {
	return "voice_files"
}
